include/linux/rcupdate.h

```diff
@@ -674,12 +674,14 @@ static inline void rcu_preempt_sleep_check(void)
 #else /* CONFIG_KTSAN */
 #define rcu_assign_pointer(p, v)				\
 	do {							\
+		typeof(*(v)) __force __rcu * ___v1 =		\
+					RCU_INITIALIZER(v);	\
 		/* hlist_bl_set_first_rcu changes the last bit	\
 		   of the pointer by applying LIST_BL_LOCKMASK.	\
 		   Do & ~7 to ignore that. */			\
 		ktsan_sync_release(				\
-			(void *)((unsigned long)(v) & ~7));	\
-		smp_store_release(&p, RCU_INITIALIZER(v));	\
+			(void *)((unsigned long)(___v1) & ~7));	\
```

Comment:
this is weird, we are doing release on the stored value rather then on &p
the same in rcu_dereference
we need to understand why we are doing it and add a comment